src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -382,13 +382,21 @@ static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2)
 *
 * emptyAttributeValue -
 *
-* Check that the attribute doesn't have any value, taking into account its multi-valuated
-* nature
+* Check that the attribute doesn't have any value, taking into account its multi-valuated nature.
+*
+* NOTE
+*   1. ValueTypeNumber and ValueTypeBoolean are always non-empty.
+*   2. In API version 2, the value of a ContextAttribute CANNOT be left out
```

Comment:
Not sure if I understand this... it could be left out if the value is not used at all. I mean:

{
    "type": "foo"
}

is a case in which emptyAttributeValue() should return true, while:

{
   "value": ""
}

is a case in which emptyAttributeValue() should return true in v2 but false in v1.